src/Microsoft.Data.Analysis/PrimitiveColumnContainer.cs

```diff
@@ -251,20 +334,20 @@ private byte SetBit(byte curBitMap, int index, bool value)
             if (value)
             {
                 newBitMap = (byte)(curBitMap | (byte)(1 << (index & 7))); //bit hack for index % 8
-                if (_modifyNullCountWhileIndexing && ((curBitMap >> (index & 7)) & 1) == 0 && index < Length && NullCount > 0)
+                if (((curBitMap >> (index & 7)) & 1) == 0 && index < Length && NullCount > 0)
```

Comment:
Can we leverage the null checker from the bitmap helper you wrote?

Maybe analyze perf and try to force it to inline --- if we think it'll be impactful?

Reply:
Fixed